ui-modules/app-inspector/app/components/entity-tree/entity-tree.directive.js

```diff
@@ -130,9 +126,10 @@ export function entityNodeDirective() {
         });
     }
 
-    function controller ($scope, $state, $stateParams) {
+    function controller ($scope, $state, $stateParams, iconService) {
         $scope.isOpen = true;
-
+        iconService.get($scope.entity, true).then(value => $scope.iconUrl = value);
```

Comment:
Should set the `entity.iconUrl` instead of adding this into the scope, so the icon will be saved on the model which is propagated everywhere.